inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/detection_output/detection_output_kernel_ref.cpp

```diff
@@ -0,0 +1,231 @@
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2});
+    } else if (idx == 1) {
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2});
+    } else if (idx == 2) {
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 2});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 1});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2});
+    } else if (idx == 3) {
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 2});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 1});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2});
+    }
+}
+
+KernelsData DetectionOutputKernelRef::GetKernelsData(const Params& params, const optional_params& options) const {
+    assert(params.GetType() == KernelType::DETECTION_OUTPUT && options.GetType() == KernelType::DETECTION_OUTPUT);
+
+    constexpr size_t kKernelsNum = 4;
+    KernelData kd = KernelData::Default<detection_output_params>(params, kKernelsNum);
+    const detection_output_params& detectOutParams = static_cast<const detection_output_params&>(params);
+
+    constexpr size_t prior_box_size = 4;
+    auto num_of_images = detectOutParams.inputs[0].Batch().v;
+    auto loc_feature_num = detectOutParams.inputs[0].Feature().v;
+    auto num_classes = detectOutParams.detectOutParams.num_classes;
+    auto num_loc_classes = (detectOutParams.detectOutParams.share_location) ? 1 : num_classes;
+    auto num_prior_boxes = (loc_feature_num / (num_loc_classes * prior_box_size));
+
+    constexpr size_t buffer_bytes = 16; // bboxes[xmin, ymin, xmax, ymax], scores[batchId, classId, boxId, score]
+    size_t buffer_stride = num_prior_boxes * buffer_bytes;
+    size_t buffer1_size = num_of_images * num_classes * buffer_stride;
+    size_t buffer2_size = num_of_images * num_classes * buffer_stride;
+    size_t buffer3_size = num_of_images * (num_classes + 1) * 4;
+
+    kd.internalBufferSizes.push_back(buffer1_size);
+    kd.internalBufferSizes.push_back(buffer2_size);
+    kd.internalBufferSizes.push_back(buffer3_size);
+    kd.internalBufferDataType = Datatype::F32;
```

Comment:
is this fixed?

Reply:
I'm not sure but when referring to other kernels(e.g. MVN) that use intermediate buffer it declared like this.